src/mapreduce.jl

```diff
@@ -66,3 +70,21 @@ function Base.count(f, v::AbstractDiskArray)
         count(f, v[chunk...])
     end
 end
+
+for (_fname, init, acum) in ((:_sum, :zero, :+), (:_prod, :one, :*),
+                             (:_all, _->:true, :&), (:_any, _->:false, :|),
+                             (:_maximum, :typemin, :max), (:_minimum, :typemax, :min))
+	@eval Base.$_fname(a::AbstractDiskArray, dims)  = Base.$_fname(identity, a, dims)
+	@eval function Base.$_fname(f::Function, a::AbstractDiskArray, dims)
+		_dims = typeof(dims)<:Tuple ? [dims...] : typeof(dims)<:Number ? [dims] : dims
+		out_dims = [size(a)...]
+		out_dims[_dims] .= 1
```

Comment:
```julia
out_dims = ntuple(Val{N}) do i
    i in _dims ? 1 : size(a)[i]
end 
```

Something like this should be the same but type stable for splats. `N` needs a `where N` on the array ndims type parameter.

Reply:
added a commit with this code.  the `out` array is still type unstable.  not sure how to fix that given it's eltype is calculated.